app/models/concerns/discourse_translator/translatable.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module DiscourseTranslator
+  module Translatable
+    extend ActiveSupport::Concern
+
+    prepended do
+      has_many :translations,
+               class_name: "DiscourseTranslator::#{name}Translation",
+               dependent: :destroy
+      has_one :content_locale, class_name: "DiscourseTranslator::#{name}Locale", dependent: :destroy
+    end
+
+    def set_detected_locale(locale)
+      # locales should be "en-US" instead of "en_US" per https://www.rfc-editor.org/rfc/rfc5646#section-2.1
+      locale = locale.to_s.gsub("_", "-")
```

Comment:
Nitpicking, but do we expect more than one `_` to `-` substitution? If no, maybe just `.sub` is enough?